src/bit_machine/exec.rs

```diff
@@ -446,29 +447,29 @@ impl BitMachine {
 
 /// Errors related to simplicity Execution
 #[derive(Debug)]
-pub enum ExecutionError<'a> {
+pub enum ExecutionError {
     /// Reached a fail node
     ReachedFailNode,
     /// Reached a pruned branch
     ReachedPrunedBranch,
-    /// Application-related error
-    AppError(Box<dyn AppError + 'a>),
+    /// Jet failed during execution
+    JetFailed,
```

Comment:
In 6ec8e71cee1906a73fb3dc5c2601e15cd6dc256c:

Why don't we wrap the `JetFailed` struct here?

Reply:
I wasn't sure if wrapping an empty struct makes sense, but we can do it.

Reply:
It's zero-cost, avoids some slightly ugly code, and it'll work if the struct ever stops being empty

Reply:
Sounds reasonable. It's in the latest commits.